Source/FindInBlueprintExternal/Private/FiBResultsToJsonWriter.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) Coconut Lizard Limited. All rights reserved.
+
+#include "FindInBlueprintExternal/Classes/FiBResultsToJsonWriter.h"
+#include <Runtime/Core/Public/HAL/FileManager.h>
+#include <Runtime/Core/Public/Misc/DateTime.h>
+#include <Runtime/Core/Public/Misc/Paths.h>
+#include <Runtime/Core/Public/Serialization/Archive.h>
+
+
+FiBResultsToJsonWriter::FiBResultsToJsonWriter(FArchive& Archive)
+	: JsonWriter(TJsonWriterFactory<TCHAR, TPrettyJsonPrintPolicy<TCHAR>>::Create(&Archive))
+{
+
+}
+
+FiBResultsToJsonWriter::~FiBResultsToJsonWriter()
+{
+	JsonWriter->Close();
+}
+
+void FiBResultsToJsonWriter::WriteDataToJson(const TArray<FSearchResult>& SearchResults, bool IsRoot)
+{
+
+	if (SearchResults.Num() <= 0)
+	{
+		return;
+	}
```

Comment:
It's really up to personal preference, though I find flipping this if statement often reads better and mandates your requirements better. In the current state this is saying "we have a problem if there are 0 or less items", which is partly always true, there's a serious issue if you have negative items in an array.
I like to read this as a requirement, so you ask if there are more than 0 items in the array, and then encapsulate your logic in it. As so:
```suggestion
	if (SearchResults.Num() > 0)
	{
		DoStuff();
	}
```